### PR TITLE
Add the capability of rewrite implicit boolean comparisons to explicit comparisons

### PR DESCRIPTION
**Pull request by alfonsohdez08**
This PR intention is to address the issue [#30185](https://github.com/Azure/azure-sdk-for-net/issues/30185). The idea is rewrite expressions such as `ent => ent.BooleanProperty`  to `ent => ent.BooleanProperty == true` so the OData expression writer produces a valid filter. This rewrite not only work when accessing an object’s boolean property, but also when calling the `TableEntity.GetBoolean` method: `ent => ent.GetBoolean("BoolProperty").Value` would get rewritten to `ent => ent.GetBoolean("BoolProperty").Value == true`. I call that this proposal cover cases where type casting is part of an expression: `ent => (bool)ent.GetBoolean("BoolProperty")`.

### CODE REVIEW

**Comment by christothes** [comment]
Thanks for the contribution, @alfonsohdez08! I'll take a look in a bit.

File: sdk/tables/Azure.Data.Tables/src/Queryable/ExpressionNormalizer.cs [hunk 1]
```diff
@@ -144,6 +188,11 @@ internal Expression VisitMethodCallNoRewrite(MethodCallExpression call)
 
             if (ReflectionUtil.s_dictionaryMethodInfosHash.Contains(visited.Method) && visited.Arguments.Count == 1 && visited.Arguments[0] is ConstantExpression)
             {
+                if (visited.Method.ReturnType == typeof(bool?) && IsImplicitBooleanComparison(visited))
+                {
+                    return CreateExplicitBooleanComparison(visited);
+                }
+
```
Comment (christothes) [inline sdk/tables/Azure.Data.Tables/src/Queryable/ExpressionNormalizer.cs:195]: I don't this this code is ever reached

**Review by christothes — COMMENTED** [review]
Looks great! Just a minor comment. 

Would you mind also adding a changelog entry for this?

File: sdk/tables/Azure.Data.Tables/src/Queryable/ExpressionNormalizer.cs (hunk 1 above)
Comment (alfonsohdez08) [reply sdk/tables/Azure.Data.Tables/src/Queryable/ExpressionNormalizer.cs]: Oh yeah, that's right! I missed revisiting this piece of code after I performed the changes for rewrite expressions that at least has a type casting expression on it. That said, the type casting rewrites will cover the scenario when having expressions such as `ent => (bool)ent.GetBoolean("BoolProp")`.

**Comment by alfonsohdez08** [comment]
Thank you, @christothes! And yeah, absolutely! I just updated the PR with the changelog entry as well.

**Review by christothes — APPROVED** [review]

**Comment by christothes** [comment]
@alfonsohdez08  One more thing - Unfortunately, I missed this as part of the release that went out today, so would you mind updating the changelog to move this fix into the latest upcoming version section? Thanks!

**Comment by alfonsohdez08** [comment]
@christothes yeah, sure! I just updated the changelog.

**Pull request merged**